### ISSUE TITLE
[BUG] When using navigateWithinModal kirby-page-title is not read by VoiceOver on the modals shown

### ISSUE THREAD

**Issue by chi-jyskebank-dk** [kirbydesign/designsystem#3801]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
When I open a modal using "navigateToModal" the kirby-page-title is read by a11y as expected. 

However, when using "navigateWithinModal" to navigate to another modal, the kirby-page-title of the new modal is not read by a11y. Instead VoiceOver reads "Netbank, Webcontent" which doesn't tell the user anything about the actual modal content shown. 

## Describe how to reproduce the bug
1. Login in to Staging Jyskebank Netbank
2. Select "Profil" from Netbank Menu
3. Select "Beløbsgrænse pr. dag"
4. Press "Skift beløbsgrænse" button - the title "Vælg din beløbsgrænse" is read by VoiceOver
5. Adjust slider to an amount higher that the actual (and at least > 50.000 kr. ie. "Højere end anbefalet")
6. Press "OK" button - the title "Vælg periode" is NOT read by VoiceOver as expected (instead it reads out "Netbank, Webindhold")
7. Press "Fortsæt >" button - the title "Obs!!" is NOT read by VoiceOver as expected (instead it reads out "Netbank, Webindhold")

## Which Kirby version was used?
10.2.1

## What was the expected behavior?
Navigation is done by calls to navigateToModal / navigateWithinModal.
I expect that title of the modals shown are read out by VoiceOver, so that the user gets an idea about where in the flow he/she is.

## Add any screenshots
https://github.com/user-attachments/assets/f22f6b30-063f-4ef6-bab8-41837be052ba

## Please complete the following information:
- OS: macOS 15.2
- Browser Chrome

## Are there any additional context?
n/a

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/develop/.github/SUPPORT.md#team-kirby); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

**Comment by RasmusKjeldgaard** [comment]
Closing this as duplicate of #3896, as the issue is not only related to routed modal flows, but also components with 'switch' like replacement of components inside the modal, e.g. stepping through a series of steps in a non-routed modal.
